nxcomp/Loop.cpp

```diff
@@ -9340,6 +8631,100 @@ int ParseCommandLineOptions(int argc, const char **argv)
 
           return -1;
         }
+        case 'd':
+        {
+          if ( argi+1 >= argc )
+          {
+            PrintUsageInfo(nextArg, 0);  
+            return -1;
+          }
+
+          int level = atoi(argv[argi+1]);
+          nx_log.level( (NXLogLevel)level );
+
+          argi++;
+          break;
+
+        }
+        case 'o':
+        {
+          if ( argi + 1 >= argc )
+          {
+            PrintUsageInfo(nextArg, 0);
+            return -1;
+          }
+
+          std::ofstream *logfile = new std::ofstream();
+
+          // Unbuffered output
+          logfile->rdbuf()->pubsetbuf(0, 0);
+          logfile->open(argv[argi+1], std::ofstream::app);
+
+          if ( logfile->is_open() )
+          {
+            nx_log.stream(logfile);
+          }
+          else
+          {
+            cerr << "Failed to open log file " << argv[argi+1] << endl;
+            return -1;
+          }
+            
+          argi++;
+          break;
+        }
+        case 'f':
+        {
+          if ( argi + 1 >= argc )
+          {
+            PrintUsageInfo(nextArg, 0);
+            return -1;
+          }
+
+          const char *format = argv[argi+1];
+          size_t pos = 0;
+
+          nx_log.log_level(false);
+          nx_log.log_time(false);
+          nx_log.log_unix_time(false);
+          nx_log.log_location(false);
+          nx_log.log_thread_id(false);
+
+          for(pos =0;pos<strlen(format);pos++)
+          {
+            switch(format[pos])
+            {
+              case '0': break;
+              case 't': nx_log.log_time(true); break;
+              case 'u': nx_log.log_time(true); nx_log.log_unix_time(true); break;
+              case 'l': nx_log.log_level(true); break;
+              case 'T': nx_log.log_thread_id(true); break;
+              case 'L': nx_log.log_location(true); break;
+              default : cerr << "Unrecognized format specifier: " << format[pos] << endl; break;
+            }
+          }
+
+          argi++;
+          break;
+        }
+        case 'b':
```

Comment:
My comment was only targeted at the `b` option, making logging configurable naturally is a good idea! Since you removed all breakpoint code, this comment is obsolete.